File: problem2.py
import itertools

def fibs(x=1, y=1):
    while True:
        yield x
        x, y = y, x+y

def main():
    fibs4m = itertools.takewhile(lambda x: x < 4000000, fibs(1, 2))
    print(sum(x for x in fibs4m if x % 2 == 0))

main()

File: problem2.hs
fibs' :: Integer -> Integer -> [Integer]
fibs' x y = x + y : fibs' y (x+y)
fibs = [1, 2] ++ fibs' 1 2

main = putStrLn $ show $ sum $ filter (even) $ takeWhile (\x -> x < 4000000) fibs
